src/Text/Pandoc/Readers/Markdown.hs

```diff
@@ -688,9 +688,13 @@ codeBlockFenced = try $ do
   rawattr <-
      (Left <$> (guardEnabled Ext_raw_attribute >> try rawAttribute))
     <|>
-     (Right <$> option ("",[],[])
-         ((guardEnabled Ext_fenced_code_attributes >> try attributes)
-          <|> ((\x -> ("",[toLanguageId x],[])) <$> many1Char nonspaceChar)))
+     (Right <$> (do
+           languageId <- option Nothing (Just . toLanguageId <$> try (many1Char $ satisfy (\x -> (x /= '{') && not (isSpace x))))
```

Comment:
Hm, I'm not really sure about this.
The current convention seems pretty reasonable in my opinion.
If you'd want to choose a more lax specification, then maybe Pandoc should follow [the Commonmark spec](https://spec.commonmark.org/0.30/#info-string).
CommonMark basically allows all characters except whitespace and, if backtick fences are used, backticks.

So, these would be valid:
````
```js
```
````
````
```c#
```
````
````
``` c#$§+!"
```
````
````
~~~c```
~~~
````

While this one wouldn't:
````
```c`
```
````

I feel like this should be a separate PR/discussion though.

Reply:
At the very least we should exclude backticks.
````
```ab```{.class}
````
should be parsed as inline code.

Commonmark spec is irrelevant here; this was a compromise because others weren't willing to commit to a particular attribute syntax, but pandoc does have one.

I'm okay with leaving the main issue for another occasion, but we should exclude backticks at least.

Reply:
Okay, I'm going to disallow the following characters: ``\`{}`` and whitespace

I don't know if there's a good way to add negative test cases for when these should fail, but I can at least add a test case for the inline code example you showed above.

Reply:
Sounds good.  Note that there's no "failed parse" for markdown; you just get different output.  So it's entirely possible to add a test case like
````
``` foo}{.bar}
test
```
````
If the parser doesn't recognize this as a fenced code block, it will fall back to interpreting it as inline code with contents `foo}{.bar}
test`

Reply:
Done :)

Shall I add support for `c#` to `csharp` in this pr or was that just an example for a later time?